testing/testing.rst

```diff
@@ -0,0 +1,842 @@
+can verify a few of our most important system requirements. However,
+when the test fails, it can sometimes be harder to track down the
+cause for the problem. (To debug a failed ERP test, you can run the
+same configuration in an ERS, PEM and/or PET test.)
+
+``ERI``:
+**Tests a combination of hybrid/branch/exact restart test**
+The ERI test carries out the following 3 runs:
+(1) Do an initial run and writing restarts with short term archiving
+on. (2) Do a hybrid run with ref1 restarts and write additional
+restarts with short term archiving on.
+(3) Do a branch run starting from restarts written in (2).
+
+Other available tests are listed below.
+
+``ERR``:
+**Short term archiving test**.
+This is an ERS test except that after the initial run the short term
+archive tool is run which moves model output out of the run directory
+into the short-term archive directory then the restart run is staged
+from the short term archive directory.
+
+``REP``:
+**Reproducibility test**:
+Do two identical runs and verify if they give the same results.
+
+``PET``:
+**Modified Threading OpenMP bit-for-bit test**.
+(1) Do an initial run where all components are threaded and add a suffix
+of `base` to the output files.  (20 Do another initial run with nthrds=1
+for all components and add a suffix of `single_thread` to the output
+files.  Compare `base` and `single_thread` output files.
+
+``PEM``:
+**Modified PE counts MPI bit-for-bit test**.
+(10 Do an initial run with default PE layout and add a suffix of `base` to
+the output files. (2) Do another initial run with modified PES
+(NTASKS_XXX => NTASKS_XXX/2) and add a suffix of `modpes` to the
+output fiels.  Compare the `base` and `modpes` output files.
+
+``PEA``:
+**Single PE test of MPI-SERIAL versus MPI**
+(1) Do an initial run on 1 pe with MPI and add a suffix of `base` to
+the output files. (2) Do the same run on 1 pe with MPI-SERIAL and add
+a suffix of `mpiserial` to the output files.  Compare the `base` and
+`mpiserial` output files.
+
+``SEQ``
+**Different PE-layout sequencing test**:
+(1) Do an initial run test with out-of-box PE-layout and add a `base`
+suffix to the output files. (2) Do a second run where all root pes are
+at pe-0 and add a `seq` suffix to the output files.  Compare the
+`base` and `seq` output files.
+
+``PFS``:
+**System performance test**.
+Do 20 day run with no restart files created.
+
+
+Common test options
+===================
+
+The following are the most commonly used test options (optional strings
+appearing after the test type, separated by ``_``):
+
+``_D``: Compile in debug mode. Exactly what this does depends on the
+compiler. Typically, this turns on checks for array bounds and various
+floating point traps. The model will run significantly slower with this
+option.
+
+``_L``:
+Specifies the length of the run. The default for most tests is 5
+days. Examples are ``_Ld3`` (3 days), ``_Lm6`` (6 months), and
```

Comment:
```suggestion
days. Examples are ``_Ln9``, ``_Ld3`` (3 days), ``_Lm6`` (6 months), and
```
Adding this because it shows up in some testlist files (e.g., CAM).